diff --git a/lib/util/version.js b/lib/util/version.js
index c9dc683b7d149c5a3a941bdcbd32f657facdfbf6..5b8f184bf865c6fd9cc4e229880121771298f150 100644
--- a/lib/util/version.js
+++ b/lib/util/version.js
@@ -28,7 +28,7 @@ function resetDetectedVersion() {
 
 function resolveBasedir(contextOrFilename) {
   if (contextOrFilename) {
-    const filename = typeof contextOrFilename === 'string' ? contextOrFilename : contextOrFilename.getFilename();
+    const filename = typeof contextOrFilename === 'string' ? contextOrFilename : (contextOrFilename.filename ?? contextOrFilename.getFilename?.());
     const dirname = path.dirname(filename);
     try {
       if (fs.statSync(filename).isFile()) {
